src/hypothesis/internal/conjecture/engine.py

```diff
@@ -43,6 +43,7 @@
 
 
 HUNG_TEST_TIME_LIMIT = 5 * 60
+MAX_SHRINKS = 500
```

Comment:
Please use 5x the max-examples value? 500 shrinks on some of my tests would be disastrous.

Reply:
> @bukzor, what's your use-case for this?

Since timeout is going away, deadline and max-examples are my only knobs for controlling runtime of my tests. Under our (corporate) test system, I have to ensure that the test completes before a hard deadline, sometimes as low as sixty seconds, and some of my use cases run for many seconds per trial. So you can see, if hypothesis were to attempt 500 shrinks, I would never see any result. A somewhat non-minimal result is much preferred.

Currently, I have these tests set to max-trials:5 max-shrinks:2, and they reliably meet deadline. Setting max-shrinks to a constant 500 will cause the test to timeout and never show any counter example upon failure.

Perhaps I'm thinking about this incorrectly, and instead I should open a feature request for an overall test deadline (currently we only have a per-trial deadline), upon which hypothesis will return what it has as quickly as possible.

Reply:
May or may not be relevant: Our tests run in a remote cluster, and I see timings that easily vary between 2-5x slower than my local machine, and I don't believe I can do anything about it. Currently, I have to set my max-examples, max-shrinks to come under budget for the *worst* machine that the test *may* run on.